### ISSUE TITLE
Reduce the amount of information returned by HTTP API

### ISSUE THREAD

**Issue by trustin** [line/centraldogma#241]
TL;DR - We are returning more information than necessary for some operations.

### Pushing a commit

I find `author`, `commitMessage` and `entries` are redundant information because `commitMessage` and `entries` are given by the client and the client should know who it is.

```json
{
  "revision": ...,
  "author": ...,
  "pushedAt": ...,
  "commitMessage": ...,
  "entries": ...
}
```

Ideally, the response could be:

```json
{
  "revision": ...,
  "pushedAt": ...
}
```

### Getting the commit list

`entries` is unnecessary. A user should send an additional query to retrieve the snapshot of a revision or the diff between two revisions based on what he or she needs.

```json
[
    {
        "revision": ...,
        "author": ...,
        "pushedAt": ...,
        "commitMessage": ...,
        "entries": [ ... ]
    },
    ...
]
```

### Watching a file

`entries` always contains a single entry. Perhaps we should just replace `entries` with `entry`?

```json
{
   "revision": ...,
   "author": ...,
   "pushedAt": ...,
   "commitMessage":, ...
   "entries": [ ... ]
}
```

### Watching a repository

`entries` could be removed, given that a user knows what files will be in the list given that he or she specified `pathPattern` parameter? Also, `entries` can increase the amount of the traffic.
